File: app/src/main/java/com/motivation/team3/motivateme/activity/UpDateNote.java
package com.motivation.team3.motivateme.activity;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import com.motivation.team3.motivateme.R;
import com.motivation.team3.motivateme.database.TaskDbHelper;
import com.readystatesoftware.systembartint.SystemBarTintManager;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;


public class UpDateNote extends AppCompatActivity
{
    EditText title,body;
    String id,stitle,sbody,time,date;
    TaskDbHelper db;
    protected void onCreate( Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.node_add_layout);

        Toolbar toolbar=(Toolbar)findViewById(R.id.note_appToolbar);
        setSupportActionBar(toolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDefaultDisplayHomeAsUpEnabled(true);

        title=(EditText)findViewById(R.id.note_title);
        body=(EditText)findViewById(R.id.note_body);

        db=new TaskDbHelper(this);

        Intent intent=getIntent();
        id=(String)intent.getSerializableExtra("ID");
        stitle=(String)intent.getSerializableExtra("TITLE");
        sbody=(String)intent.getSerializableExtra("BODY");
        time= (String)intent.getSerializableExtra("TIME");
        date=(String)intent.getSerializableExtra("DATE");

        title.setText(stitle);
        int textlength=title.getText().length();
        body.setText(sbody);
        int bodylength=body.getText().length();

        if(textlength!=0)
        {
            title.setSelection(textlength,textlength);
        }
        else if(textlength==0 && bodylength!=0)
        {
            body.setSelection(bodylength,bodylength);
        }

        SystemBarTintManager tintManager = new SystemBarTintManager(this);

        tintManager.setStatusBarTintEnabled(true);
        tintManager.setStatusBarTintColor(Color.parseColor("#0288d1"));
        tintManager.setNavigationBarTintEnabled(true);
    }

    public boolean onCreateOptionsMenu(Menu menu)
    {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    public boolean onOptionsItemSelected(MenuItem item)
    {
        switch (item.getItemId())
        {
            case R.id.action_add_task:
                String stext=title.getText().toString();
                String sbody=body.getText().toString();
                if(title.getText().toString().length()!=0 || body.getText().toString().length()!=0)
                {
                    Calendar now = Calendar.getInstance();
                    Date dt = new Date(0, 0, 0, now.get(Calendar.HOUR_OF_DAY), now.get(Calendar.MINUTE), now.get(Calendar.SECOND));
                    SimpleDateFormat sdf = new SimpleDateFormat("hh:mm a");
                    String timeText = sdf.format(dt);
                    String dateText = String.valueOf(now.get(Calendar.DAY_OF_MONTH)) + "/" + String.valueOf(now.get(Calendar.MONTH) + 1) + "/" + String.valueOf(now.get(Calendar.YEAR));

                    db.updateNoteData(Integer.parseInt(id), title.getText().toString(), body.getText().toString(), timeText,dateText);
                }
                else if(stext.length()==0 || sbody.length()==0)
                {
                    db.deleteNoteData(Integer.parseInt(id));
                }
                finish();
                return true;
        }
        finish();
        return true;
    }
}

File: app/src/main/java/com/motivation/team3/motivateme/activity/UpDateNote.kt
package com.motivation.team3.motivateme.activity

import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import android.widget.EditText
import com.motivation.team3.motivateme.R
import com.motivation.team3.motivateme.database.TaskDbHelper
import com.readystatesoftware.systembartint.SystemBarTintManager
import java.text.SimpleDateFormat
import java.util.*

class UpDateNote : AppCompatActivity() {
    var title: EditText? = null
    var body: EditText? = null
    var id: String? = null
    var stitle: String? = null
    var sbody: String? = null
    var time: String? = null
    var date: String? = null
    var db: TaskDbHelper? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.node_add_layout)
        val toolbar = findViewById(R.id.note_appToolbar) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeButtonEnabled(true)
        supportActionBar!!.setDefaultDisplayHomeAsUpEnabled(true)
        title = findViewById(R.id.note_title) as EditText
        body = findViewById(R.id.note_body) as EditText
        db = TaskDbHelper(this)
        val intent = intent
        id = intent.getSerializableExtra("ID") as String?
        stitle = intent.getSerializableExtra("TITLE") as String?
        sbody = intent.getSerializableExtra("BODY") as String?
        time = intent.getSerializableExtra("TIME") as String?
        date = intent.getSerializableExtra("DATE") as String?
        title!!.setText(stitle)
        val textlength = title!!.text.length
        body!!.setText(sbody)
        val bodylength = body!!.text.length
        if (textlength != 0) {
            title!!.setSelection(textlength, textlength)
        } else if (textlength == 0 && bodylength != 0) {
            body!!.setSelection(bodylength, bodylength)
        }
        val tintManager = SystemBarTintManager(this)
        tintManager.isStatusBarTintEnabled = true
        tintManager.setStatusBarTintColor(Color.parseColor("#0288d1"))
        tintManager.setNavigationBarTintEnabled(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_add_task -> {
                val stext = title!!.text.toString()
                val sbody = body!!.text.toString()
                if (title!!.text.toString().length != 0 || body!!.text.toString().length != 0) {
                    val now = Calendar.getInstance()
                    val dt = Date(
                        0,
                        0,
                        0,
                        now[Calendar.HOUR_OF_DAY],
                        now[Calendar.MINUTE],
                        now[Calendar.SECOND]
                    )
                    val sdf = SimpleDateFormat("hh:mm a")
                    val timeText = sdf.format(dt)
                    val dateText =
                        now[Calendar.DAY_OF_MONTH].toString() + "/" + (now[Calendar.MONTH] + 1).toString() + "/" + now[Calendar.YEAR].toString()
                    db!!.updateNoteData(
                        id!!.toInt(),
                        title!!.text.toString(),
                        body!!.text.toString(),
                        timeText,
                        dateText
                    )
                } else if (stext.length == 0 || sbody.length == 0) {
                    db!!.deleteNoteData(id!!.toInt())
                }
                finish()
                return true
            }
        }
        finish()
        return true
    }
}